corehq/apps/app_manager/suite_xml.py

```diff
@@ -1943,6 +1944,46 @@ def get_target_module(case_type, module_id, with_product_details=False):
             except IndexError:
                 pass
 
+        self.add_parent_datums(datums, module)
+
+        return datums, assertions
+
+    def add_parent_datums(self, datums, module):
+
+        def update_refs(action_, datum_, changed_ids):
+            if action_:
+                # Only advanced module actions have a parent_tag attribute. Basic modules don't need one
+                # because they only deal with one case type. Set parent_tag to None for them.
+                parent_tag = getattr(action_, 'parent_tag', None)
+                if parent_tag in changed_ids:
+                    # update any reference to previously changed datums
+                    change = changed_ids[parent_tag]
+                    nodeset = datum_.nodeset
+                    old = session_var(change['old_id'])
+                    new = session_var(change['new_id'])
+                    datum_.nodeset = nodeset.replace(old, new)
+
+        def avoid_duplicate_ids(action_, this_datum_, parent_datum_, datum_ids_, changed_ids):
+            if action_:
+                case_tag = getattr(action_, 'case_tag', None)  # Set case_tag to None for basic modules
```

Comment:
Yes, your assumptions are correct its just that for someone who doesn't
know that this code doesn't make sense (or looks like it could be a bug). I
was also wondering if we can't just treat basic modules differently since
this whole algorithm seems overkill.